cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -976,6 +980,33 @@ public String getVaultNameFromEncryptionKeyUrl(String encryptionKeyUrl) {
         return null;
     }
 
+    public Map<String, List<String>> getAvailabilityZones(String region) throws ProviderAuthenticationFailedException {
+        return handleException(() -> {
+            Map<String, List<String>> zoneInfo = new HashMap<>();
+            if (StringUtils.isEmpty(region)) {
+                return zoneInfo;
+            }
+            String criteria = "location eq '" + RegionUtil.findByLabelOrName(region).name() + "'";
+            LOGGER.debug("Fetch AZ info from Azure for region {} and criteria {}", region, criteria);
+            AzureListResult<ResourceSkuInner> azureResult = azureListResultFactory.create(azure
+                    .virtualMachines()
+                    .manager()
+                    .serviceClient()
+                    .getResourceSkus()
```

Comment:
It is a very expensive call, might worth caching? The raw JSON for all of the regions returned is about 60MB as of now...

Reply:
It's already being cached when VM information is fetched.
@Cacheable(cacheNames = "cloudResourceVmTypeCache", key = "#cloudCredential?.id + #region.getRegionName()")
    public CloudVmTypes virtualMachines(ExtendedCloudCredential cloudCredential, Region region, Map<String, String> filters)